pins/boards.py

```diff
@@ -326,7 +313,55 @@ def pin_write(
 
         return meta
 
-    def pin_download(self, name, version=None, hash=None):
+    def pin_write(
+        self,
+        x,
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Mapping] = None,
+        versioned: Optional[bool] = None,
+        created: Optional[datetime] = None,
+    ) -> Meta:
+        """Write a pin object to the board.
+
+        Parameters
+        ----------
+        x:
+            An object (e.g. a pandas DataFrame) to pin.
+        name:
+            Pin name.
+        type:
+            File type used to save `x` to disk. May be "csv", "arrow", "parquet",
+            "joblib", "json", or "file".
```

Comment:
Can we remove `file` as a pin type if it is going to raise an error?

```suggestion
            "joblib", or "json".
```

Reply:
Ah, removing it makes sense! With `pin_upload`, it seems like the `type="file"` is now of more an internal detail (since it's set automatically with `pin_upload`)